cmake/luajit.cmake

```diff
@@ -69,14 +69,10 @@ endif()
 # Enable LuaJIT ASan support. The internal LuaJIT memory allocator
 # is not instrumented yet unfortunately, so to find any memory
 # faults it's worth building LuaJIT with system provided memory
-# allocator (i.e. enable LUAJIT_USE_SYSMALLOC option). However,
-# Tarantool doesn't finalize Lua universe the right way (see the
-# comments near <tarantool_lua_free>), so running Tarantool
-# testing routine with LUAJIT_USE_SYSMALLOC enabled generates
-# false-positive LSan leaks. Return back here to enable
-# LUAJIT_USE_SYSMALLOC, when the issue below is resolved.
-# https://github.com/tarantool/tarantool/issues/3071
+# allocator (i.e. enable LUAJIT_USE_SYSMALLOC option).
 if(ENABLE_ASAN)
+    set(LUAJIT_USE_SYSMALLOC ON CACHE BOOL
```

Comment:
For non-GC64 builds we can't use sysalloc since the bitness of the allocator and pointers should be the same. See [the following build failure](https://github.com/tarantool/tarantool/actions/runs/10793652440/job/29936087657?pr=10554#step:8:3760).

I suppose we should enable this flag only when the `LUAJIT_ENABLE_GC64` is given as well.